src/main/java/minijava/Cli.java

```diff
@@ -267,10 +277,9 @@ static Parameters parse(String... args) {
       Parameters params = new Parameters();
       try {
         new JCommander(params, args);
-        if (params.mainParameters.size() == 1) {
-          params.file = params.mainParameters.get(0);
-        }
+        params.file = params.mainParameters.get(params.mainParameters.size() - 1);
```

Comment:
Since you've decided to use some other workaround to let the test pass, I'll try to make this more clear:
This line is plain _wrong_. It will throw an `IndexOutOfBoundsException` if no "main argument" is was given.